### PR TITLE
Craetes a function to get the max size for abi.Argumetns, given any outermost slices have N elements

### PR DESCRIPTION
**Pull request by nolag**

### CODE REVIEW

File: core/services/relay/evm/size_helper.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	relaytypes "github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+func GetMaxSizeFormEntry(n int, entry *CodecEntry) (int, error) {
+	if entry == nil {
+		return 0, relaytypes.InvalidTypeError{}
+	}
+	return GetMaxSize(n, entry.Args)
+}
```
Comment (jmank88) [inline core/services/relay/evm/size_helper.go:14]: What about a method? Can shorten the name and drop the nil check:
```suggestion
func (e *CodecEntry) GetMaxSize(n int) (int, error) {
	return GetMaxSize(n, e.Args)
}
```

Comment (nolag) [reply core/services/relay/evm/size_helper.go:14]: sg, I'll move it into the right file.

File: core/services/relay/evm/size_helper.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		if err != nil {
+			return 0, err
+		}
+		size += argSize
+	}
+
+	return size, nil
+}
+
+func getTypeSize(n int, t *abi.Type, dynamicTypeAllowed bool, isNested bool) (int, bool, error) {
```
Comment (ilija42) [inline core/services/relay/evm/size_helper.go:30]: nit, why is abi.Type a pointer?

File: core/services/relay/evm/size_helper.go (hunk 1 above)
Comment (nolag) [reply core/services/relay/evm/size_helper.go]: Ah sorry, I must have grabbed an older version  I checked it out again after merge conflicts :(.  Will fix again.

File: core/services/relay/evm/size_helper.go (hunk 2 above)
Comment (ilija42) [reply core/services/relay/evm/size_helper.go:23]: wdym, why would you copy it by value. If you pass a dereferenced `arg.Type` into ` getTypeSize` you get the same effect of passing &tmp. Non pointer attributes inside of the struct won't get changed and attributes with a pointer would get changed either way.

**Review by jmank88 — APPROVED** [review]

File: core/services/relay/evm/size_helper.go (hunk 2 above)
Comment (nolag) [reply core/services/relay/evm/size_helper.go:23]: See below, t.Elm is passed in on line 27, it's naturally a pointer.  When you pass a struct directly, you force a copy of that struct to be made.

**Pull request merged**